extra_playbooks/upgrade-only-k8s.yml

```diff
@@ -31,6 +31,9 @@
   vars:
     ansible_ssh_pipelining: true
   gather_facts: true
+  roles:
+    - { role: kubespray-defaults }
+    - { role: etcd, tags: facts }
```

Comment:
As per my understanding this will upgrade `etcd` as well, which this playbook isn't supposed to do. How are you limiting to `facts` tags while importing `etcd` role ?

Reply:
Have a look at this, https://github.com/ansible/ansible/issues/12224

Reply:
I see how role tags work now (had not used ansible seriously before).  But interestingly if the `Set etcd_client_cert_serial` task was added with the `facts` tag but not `Gen_certs | Get etcd certificate serials`, it would still fail.  That's why I thought it's only running the ones with facts.

So I suppose to fix this properly we need to move the cert_serial stuff to a different place that can be included? or `kubespray-defaults` would be the right place?

Reply:
I had few ideas:

1. Take these tasks out in separate file in the `etcd` role and use `include_role` in `upgrade-only-k8s.yml` to include these tasks only.
2. Use the same workflow I mentioned above with `when` and some `variable`.
3. May be make `etcd` upgrade part of this playbook.

I personally think [1] should be fine but it would be great if @mattymo have some feedback!